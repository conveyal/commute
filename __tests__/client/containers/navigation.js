/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {makeMockStore, mockStores} from '../../test-utils/mock-data.js'

import Navigation from '../../../client/containers/navigation'

const mockStore = makeMockStore(mockStores.init)

describe('Container > Navigation', () => {
  it('renders correctly', () => {
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <Navigation />
      </Provider>
    )
    expect(mountToJson(tree.find(Navigation))).toMatchSnapshot()
  })
})
