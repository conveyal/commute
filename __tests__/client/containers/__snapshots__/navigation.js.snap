exports[`Container > Navigation renders correctly 1`] = `
<Connect(Navigation)>
  <Navigation
    login={[Function]}
    logout={[Function]}
    userIsLoggedIn={false}
    username={undefined}>
    <Uncontrolled(Navbar)
      bsStyle="default"
      fluid={true}>
      <Navbar
        bsClass="navbar"
        bsStyle="default"
        componentClass="nav"
        expanded={undefined}
        fixedBottom={false}
        fixedTop={false}
        fluid={true}
        inverse={false}
        onToggle={[Function]}
        staticTop={false}>
        <nav
          className="navbar navbar-default">
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={true}>
            <div
              className="container-fluid">
              <NavbarHeader>
                <div
                  className="navbar-header">
                  <NavbarBrand>
                    <a
                      className="navbar-brand"
                      href="/">
                      Commute
                    </a>
                  </NavbarBrand>
                  <NavbarToggle>
                    <button
                      className="navbar-toggle collapsed"
                      onClick={[Function]}
                      type="button">
                      <span
                        className="sr-only">
                        Toggle navigation
                      </span>
                      <span
                        className="icon-bar" />
                      <span
                        className="icon-bar" />
                      <span
                        className="icon-bar" />
                    </button>
                  </NavbarToggle>
                </div>
              </NavbarHeader>
              <NavbarCollapse>
                <Collapse
                  dimension="height"
                  getDimensionValue={[Function]}
                  in={false}
                  timeout={300}
                  transitionAppear={false}
                  unmountOnExit={false}>
                  <Transition
                    aria-expanded={null}
                    className=""
                    enteredClassName="collapse in"
                    enteringClassName="collapsing"
                    exitedClassName="collapse"
                    exitingClassName="collapsing"
                    in={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    transitionAppear={false}
                    unmountOnExit={false}>
                    <div
                      aria-expanded={null}
                      className="navbar-collapse collapse">
                      <Nav
                        bsClass="nav"
                        justified={false}
                        pullLeft={false}
                        pullRight={true}
                        stacked={false}>
                        <ul
                          className="nav navbar-nav navbar-right"
                          role={null}>
                          <NavItem
                            active={false}
                            activeHref={undefined}
                            activeKey={undefined}
                            disabled={false}
                            eventKey={1}
                            onSelect={null}>
                            <li
                              className=""
                              role="presentation"
                              style={undefined}>
                              <SafeAnchor
                                componentClass="a"
                                disabled={false}
                                onClick={[Function]}>
                                <a
                                  href=""
                                  onClick={[Function]}
                                  role="button">
                                  <Button
                                    active={false}
                                    block={false}
                                    bsClass="btn"
                                    bsStyle="default"
                                    disabled={false}
                                    onClick={[Function]}>
                                    <button
                                      className="btn btn-default"
                                      disabled={false}
                                      onClick={[Function]}
                                      type="button">
                                      Log in
                                    </button>
                                  </Button>
                                </a>
                              </SafeAnchor>
                            </li>
                          </NavItem>
                        </ul>
                      </Nav>
                    </div>
                  </Transition>
                </Collapse>
              </NavbarCollapse>
            </div>
          </Grid>
        </nav>
      </Navbar>
    </Uncontrolled(Navbar)>
  </Navigation>
</Connect(Navigation)>
`;
